### PR TITLE
Fix/merge test improvements

### LINKED ISSUE #101: Add testing
More testing is required for the different components of the project:
- [ ] frontend
- [ ] webapi
- [ ] harvester
- [ ] db-loader
- [ ] downloader
- [ ] active deployment
     - [ ] active endpoints 
     - [ ] linked download files are accessible
     - [ ] download requests are run accordingly
     - [ ] harvesters are running periodically

### PR DESCRIPTION
**Pull request by vmehraOCN**
## Overview
This branch builds a comprehensive backend test suite on top of the OBIS integration work merged from `feat/harvest-dashboard`. It resolves several integration regressions, runtime errors, and broken test paths that surfaced during the branch merge.

### ➕ New Additions
*   **Backend Test Infrastructure (`tests/`):** Added a full `pytest` suite from scratch including 113 unit tests across 10 modules (covering ERDDAP, CKAN, and compliance tracking), an end-to-end chained integration test with mocked I/O, and standalone dependency wiring.

### 🛠️ Fixes Implemented
*   **UnboundLocalError (`erddap_harvester.py`):** Fixed a critical production risk where `dataset_logger` was referenced in an HTTP error block before assignment. It is now safely pre-initialized.
*   **Prefect Context Failures:** Resolved `MissingContextError` in unit tests by implementing a `try/except` fallback to standard Python `logging.getLogger()` when run outside active Prefect flows.
*   **Prefect API Connection Failures:** Updated database loader tests to invoke the underlying function directly via `main.fn(...)` and patched loggers to bypass local Prefect API connection attempts.
*   **Legacy Imports & Legacy Test Types:** Fixed broken imports resulting from the `harvest_erddap.py` to `erddap_harvester.py` rename, and corrected a mismatched exception assertion (`ResponseTooLargeError`) in the ERDDAP client tests.

## Testing Checklist
- [x] Run backend unit and pipeline tests: `cd tests && uv run pytest`
- [x] Run backend integration smoke tests: `uv run pytest tests/integration/`

### CODE REVIEW

File: download_scheduler/download_scheduler/download_scheduler.py [hunk 1]
```diff
@@ -27,17 +28,15 @@
...
-    )
+sentry_sdk.init(
+    dsn=os.environ.get("SENTRY_DSN"),
+    integrations=[
+        LoguruIntegration(),
+    ],
+    environment=os.environ.get("ENVIRONMENT", "development"),
+    traces_sample_rate=1.0,
+    ignore_errors=[KeyboardInterrupt],
+)
```
Comment (vmehraOCN) [inline download_scheduler/download_scheduler/download_scheduler.py]: Need to review this change.

File: frontend/src/components/Controls/Filter/FilterButton/FilterButton.jsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import './styles.css'
 
 export default function FilterButton({ setOptionsSelected, optionsSelected, option }) {
   const { t, i18n } = useTranslation()
+  if (!option) return null
```
Comment (vmehraOCN) [inline frontend/src/components/Controls/Filter/FilterButton/FilterButton.jsx:11]: Review this

File: frontend/src/components/Map/Map.js [hunk 3]
```diff
@@ -74,7 +74,71 @@ export default function CreateMap({
       pitchWithRotate: false,
       dragRotate: false,
       touchZoomRotate: false
-    }
+    },
```
Comment (vmehraOCN) [inline frontend/src/components/Map/Map.js:77]: Reason for this change:
When the codebase migrated from mapbox-gl-draw to maplibre-gl-draw (commit c7cb184), the new library doesn't ship with default draw styles the way Mapbox did — you have to provide them explicitly. The follow-up commit d617d9c added a styles array to the draw tool constructor to restore the visual appearance of polygon/rectangle drawing:

Fill layer — semi-transparent fill for in-progress polygons, orange when active, blue when inactive
Line layer — dashed outline for polygon edges and lines
Point layers — white outer ring + coloured inner dot for polygon vertices and feature points, with active/inactive states
Midpoint layer — small orange dot at the midpoint of each edge (for inserting new vertices)
Without this, drawn polygons would be invisible or use broken default styles from the old Mapbox library. It's purely a visual fix tied to the library migration.

Will need to run and verify.

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -12,19 +12,19 @@ jobs:
...
-          ssh_host: ${{ vars.SERVER_IP }}
-          ssh_username: "github-runner"
-          ssh_key: ${{ secrets.AZURE_CO_RUNNER_PRIVATE_KEY }}
+          wg_config: ${{ secrets.BELUGA_WIREGUARD_VPN_CONFIG }}
+          ssh_host: "${{ secrets.SERVER_IP }}"
+          ssh_username: "runner"
+          ssh_key: ${{ secrets.BELUGA_RUNNER_PRIVATE_KEY }}
           deploy_path: "/srv/services/explore-cioos"
-          stack_name: "explore-cioos-${{ github.ref_name == 'master' && 'production' || 'development' }}"
+          stack_name: "explore-cioos-${{ (github.event.workflow_run.head_branch || github.ref_name) == 'master' && 'production' || 'development' }}"
```
Comment (vmehraOCN) [inline .github/workflows/deploy.yml:27]: Review these changes.

File: README.md [hunk 5]
```diff
@@ -148,6 +148,10 @@ For complete local development with all services running outside Docker (advance
 
 8. See website at <http://localhost:8000>
 
+## CI/CD
+
+Pushes to `master` and `development` automatically deploy to the corresponding environment via the [Deploy workflow](.github/workflows/deploy.yml). The workflow connects to the remote server over WireGuard VPN, syncs the repository to the exact commit that triggered the run, injects secrets from 1Password, and brings up the Docker Compose stack.
```
Comment (vmehraOCN) [inline README.md:153]: Is this the correct way for us to deploy at this time?

File: download_scheduler/download_scheduler/download_scheduler.py (hunk 1 above)
Comment (vmehraOCN) [reply download_scheduler/download_scheduler/download_scheduler.py]: reason for the change:
Change 1 — Sentry initialisation (lines 30–40)

The old code hard-coded a Sentry DSN as a string literal and only initialised Sentry when ENVIRONMENT == "production". The new code:

Reads the DSN from SENTRY_DSN env var — removes the exposed secret from source code
Removes the production-only guard and instead passes environment= to Sentry, which handles filtering on Sentry's side
Adds LoguruIntegration so Sentry automatically captures loguru log entries
Change 2 — Error reporting in the download failure handler (lines 185–187)

The old code called logger.error(e), logger.error(stack_trace), and then a manual sentry_sdk.capture_message(...). The new code uses logger.bind(email=..., job_id=..., pk=...).error(e), which:

Attaches structured context (who triggered it, which job, which dataset PK) directly to the log entry
Drops the explicit capture_message call — redundant now that LoguruIntegration automatically forwards loguru error() calls to Sentry

File: frontend/src/components/Controls/Filter/FilterButton/FilterButton.jsx (hunk 2 above)
Comment (vmehraOCN) [reply frontend/src/components/Controls/Filter/FilterButton/FilterButton.jsx:11]: Without it, if option is undefined or null (which can happen transiently during a render before the filter data loads, or when the parent maps over a list that includes an empty slot), the very next line option[\hover_${i18n.language}`]would throw aTypeError: Cannot read properties of undefined`. The component would crash the whole filter panel.

The fix makes FilterButton defensive — if it receives no option prop it silently renders nothing rather than throwing. All the actual button behaviour (click handler, tooltip, selected state) is unchanged.

File: download_scheduler/download_scheduler/download_scheduler.py (hunk 1 above)
Comment (vmehraOCN) [reply download_scheduler/download_scheduler/download_scheduler.py]: Resolved unless further input is required.

File: frontend/src/components/Controls/Filter/FilterButton/FilterButton.jsx (hunk 2 above)
Comment (vmehraOCN) [reply frontend/src/components/Controls/Filter/FilterButton/FilterButton.jsx:11]: Resolved unless further input is required.

File: frontend/src/components/Map/Map.js (hunk 3 above)
Comment (vmehraOCN) [reply frontend/src/components/Map/Map.js:77]: Resolved unless further input is required.

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (vmehraOCN) [reply .github/workflows/deploy.yml:27]: @JessyBarrette I would like your input on these changes. It was auto generated and I'm not sure if it should be kept. Please let me know what you think.

File: README.md (hunk 5 above)
Comment (vmehraOCN) [reply README.md:153]: This looks to be intentional and correct. Resolved unless further input is required.

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (JessyBarrette) [reply .github/workflows/deploy.yml:27]: I think those changes are because feat/harvest-dasboard has not integrated the latest changes from development. Yours is.

merging dev into feat/harvest-dasboard should resovle that.

**Comment by JessyBarrette** [comment]
I'm not a huge fan of the docs/ directory suggested here which was likely generated by AI and will more than likely drift from the active implementation. If the objectif is to help an agent to discover the project. I would ideally migrate the core to an AGENT.md or CLAUDE.md

For developpers, a readme for each modules should be more than enough. 

I'm not sure what are the objectives with the harvest_validator.

**Comment by JessyBarrette** [comment]
I would also keep the suggested changes to the frontend and api's as a sperate repos. Those could be merge right away if and do not depend on any aspect of the harvester.

**Comment by vmehraOCN** [comment]
I'll remove the docs as git tractable items and keep it just within my version of the repo. I personally find it helpful but yes, its not really needed at this time.

**Comment by vmehraOCN** [comment]
The intention with harvest_validator module was to add a way to run harvests while keeping track of logs for the purpose of report generation per harvest run. I think there has been work done on this in the form of a dashboard now but not so much built into the harvest itself yet.

If you think its redundant or not needed, I can remove this. Let me know what you think @JessyBarrette

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (vmehraOCN) [reply .github/workflows/deploy.yml:27]: I see you went ahead with that on pull 96. Tag me if you need a reviewer.

File: download_scheduler/download_scheduler/download_scheduler.py
Comment (JessyBarrette) [inline download_scheduler/download_scheduler/download_scheduler.py:1]: Not sure if those changes are coming from you or those are in development but not already in feat/harvest-dashboard, but they could be seperated if not already in development. I believe they are though

**Comment by vmehraOCN** [comment]
Okay, I removed all of the web-api changes into a separate branch this one should be able to go through now unless we have some more comments or concerns. @RichardKelly84 @JessyBarrette

**Comment by vmehraOCN** [comment]
Whoops, I removed some changes in the web-api, fixing....

**Review by vmehraOCN — COMMENTED** [review]
Okay, review done on my side.

File: harvester/uml_diagram.txt
Comment (JessyBarrette) [inline harvester/uml_diagram.txt:1]: if we want to keep such diagram I would recommand using at least a mermaid diagram.

**Comment by JessyBarrette** [comment]
@vmehraOCN could you update the title and description of the PR to reflect the actual changes suggested.

The changes
- [x] For diagrams I would suggest using  mermaid diagrams  which I believe will be simpler to maintain and visualize. 
- [x] `test/..` I would move different tests to the component for which it is related. We can still keep a test for the whole workflow. But tests for harvester would ideally remain in the harvester direction. Same for db-loader, scheduler, frontend, web-api.

This will help reducing which tests are run when.

**Comment by vmehraOCN** [comment]
Added in the changes requested @JessyBarrette.

**Comment by vmehraOCN** [comment]
Note, I left the integration test I made within harvester since its is an integration test for the harvester.
We will need to think up of some more specific tests for cross component interactions.

File: db-loader/tests/pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+[project]
+name = "db-loader-tests"
+version = "0.1.0"
+description = "Unit tests for the CIOOS db-loader"
+requires-python = ">=3.10,<3.11"
+dependencies = [
+    "db-loader",
+    "harvester",
+    "pytest>=8.2",
```
Comment (JessyBarrette) [inline db-loader/tests/pyproject.toml]: pytest-* dependancies should be added as dev dependancies

File: harvester/tests/pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+[project]
+name = "harvester-tests"
+version = "0.1.0"
+description = "Unit and integration tests for the CIOOS harvester pipeline"
+requires-python = ">=3.10,<3.11"
+dependencies = [
+    "harvester",
+    "db-loader",
+    "pytest>=8.2",
+    "pytest-mock>=3.14",
```
Comment (JessyBarrette) [inline harvester/tests/pyproject.toml]: migrate pytest* to dev dependancies

File: db-loader/tests/pyproject.toml
Comment (JessyBarrette) [inline db-loader/tests/pyproject.toml:1]: that pyproject shouldn't be there it needs to be integrated within the db-loaded/pyproject.toml  as dev dependancies

**Review by JessyBarrette — COMMENTED** [review]
## db-loader
The added pyproject.toml in db-loader/tests/pyproject.toml needs to me integrated within the main pyproject of db-loader as a dev depedancies.

## harvester
We need to migrate the pytest dependancies to dev. We dont need them in production. 

Ideally we would also need to run those during the CI/CD. The question is when and how.

**Comment by JessyBarrette** [comment]
I added some of the depandcies to the pyprojects and reviewer the original test_downloader.yaml action to run the different unit tests. Now named test-unit-tests.yaml, it will run the unit tests within each modules if changes are made within them.

File: db-loader/tests/pyproject.toml (hunk 6 above)
Comment (vmehraOCN) [reply db-loader/tests/pyproject.toml]: created dev dependencies.

File: harvester/tests/pyproject.toml (hunk 7 above)
Comment (vmehraOCN) [reply harvester/tests/pyproject.toml]: created dev dependencies.

File: db-loader/tests/pyproject.toml
Comment (vmehraOCN) [reply db-loader/tests/pyproject.toml:1]: I removed both of those pyproject.toml

File: harvester/uml_diagram.txt
Comment (vmehraOCN) [reply harvester/uml_diagram.txt:1]: This is done too.

File: download_scheduler/download_scheduler/download_scheduler.py
Comment (vmehraOCN) [reply download_scheduler/download_scheduler/download_scheduler.py:1]: I think this is addressed too unless I haven't understood what you mean here.

**Comment by vmehraOCN** [comment]
That sounds pretty good to me!

**Comment by vmehraOCN** [comment]
@RichardKelly84 @JessyBarrette I think I've addressed all current concerns on this PR. Please let me know if there is anything else to review.

File: harvester/tests/unit/test_csv_generation.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    df.to_csv(path, index=False)
+    return pd.read_csv(path)
+
+
+# ---------------------------------------------------------------------------
+# Tests: datasets.csv
+# ---------------------------------------------------------------------------
+
+class TestDatasetsCsvStructure:
+    def test_all_required_columns_present(self, sample_datasets_df):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: this will always succeed unless someone edits sample_datasets_df in conftest.py or delete a column in REQUIRED_DATASETS_COLS, so not really a useful test.

File: harvester/tests/unit/test_csv_generation.py [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        for col in REQUIRED_DATASETS_COLS:
+            assert col in sample_datasets_df.columns, f"Missing column: {col}"
+
+    def test_dataset_id_is_non_empty(self, sample_datasets_df):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: same thing, this one will always pass. not sure it is useful.

File: harvester/tests/unit/test_csv_generation.py [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+        assert sample_datasets_df["dataset_id"].notna().all()
+        assert (sample_datasets_df["dataset_id"] != "").all()
+
+    def test_eovs_is_list_after_roundtrip(self, sample_datasets_df, tmp_path):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: a bit of copy and paste code. Can be put into one instead of test_eovs_is_list_after_roundtrip,  test_organizations_is_list_after_roundtrip and test_profile_variables_is_list_after_roundtrip  . E.G:      

```
@pytest.mark.parametrize("column", ["eovs", "organizations", "profile_variables"])
      def test_array_column_is_list_after_roundtrip(self, sample_datasets_df, tmp_path, column):
          df = _write_and_read_csv(sample_datasets_df, tmp_path, "datasets.csv")
          parsed = df[column].apply(ast.literal_eval)
          assert all(isinstance(v, list) for v in parsed)
```

File: harvester/tests/unit/test_csv_generation.py [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+    def test_profile_variables_is_list_after_roundtrip(self, sample_datasets_df, tmp_path):
+        df = _write_and_read_csv(sample_datasets_df, tmp_path, "datasets.csv")
+        pv_parsed = df["profile_variables"].apply(ast.literal_eval)
+        assert all(isinstance(v, list) for v in pv_parsed)
+
+    def test_no_duplicate_dataset_ids_after_write(self, sample_datasets_df, tmp_path):
+        # Simulate the drop_duplicates call in __main__.py
+        deduped = sample_datasets_df.drop_duplicates(["erddap_url", "dataset_id"])
+        df = _write_and_read_csv(deduped, tmp_path, "datasets.csv")
+        assert df.duplicated(["erddap_url", "dataset_id"]).sum() == 0
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: This is always True, even if the test dataset has duplicates. Suggest removing this test

File: harvester/tests/unit/test_csv_generation.py [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+
+
+# ---------------------------------------------------------------------------
+# Tests: profiles.csv
+# ---------------------------------------------------------------------------
+
+class TestProfilesCsvStructure:
+    def test_all_required_columns_present(self, sample_profiles_df):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: This will always pass unless you edit the test.

File: harvester/tests/unit/test_csv_generation.py [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+        for col in REQUIRED_PROFILES_COLS:
+            assert col in sample_profiles_df.columns, f"Missing column: {col}"
+
+    def test_latitude_within_bounds(self, sample_profiles_df):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: This will always pass

File: harvester/tests/unit/test_csv_generation.py [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+        assert (sample_profiles_df["latitude"] > -90).all()
+        assert (sample_profiles_df["latitude"] < 90).all()
+
+    def test_longitude_within_bounds(self, sample_profiles_df):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: likewise, will always pass

File: harvester/tests/unit/test_csv_generation.py [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+        assert (sample_profiles_df["longitude"] > -180).all()
+        assert (sample_profiles_df["longitude"] < 180).all()
+
+    def test_depth_min_le_depth_max(self, sample_profiles_df):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: will always pass

File: harvester/tests/unit/test_csv_generation.py [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+        assert (sample_profiles_df["depth_min"] <= sample_profiles_df["depth_max"]).all()
+
+    def test_time_min_before_time_max(self, sample_profiles_df):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: will always pass

File: harvester/tests/unit/test_csv_generation.py [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+        assert (sample_profiles_df["time_min"] < sample_profiles_df["time_max"]).all()
+
+    def test_n_records_positive(self, sample_profiles_df):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_csv_generation.py]: will always pass

File: harvester/tests/unit/test_ckan.py [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+    def test_invalid_url_raises_value_error(self):
+        with pytest.raises(ValueError, match="Invalid URL format"):
+            split_erddap_url("https://example.com/not/erddap")
+
+
+# ---------------------------------------------------------------------------
+# Tests: ASCII unescaping
+# ---------------------------------------------------------------------------
+
+class TestUnescapeAscii:
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_ckan.py:75]: GAP: no test here for when there are actual escapes, for example: assert unescape_ascii(r"Caf\u00e9") == "Café"

File: harvester/tests/unit/test_profiles.py [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+    single_station_dataset.profile_ids = bad_ids
+    return single_station_dataset
+
+
+# ---------------------------------------------------------------------------
+# Tests
+# ---------------------------------------------------------------------------
+
+class TestGetProfilesHappyPath:
+    def test_returns_dataframe(self, single_station_dataset):
```
Comment (RichardKelly84) [inline harvester/tests/unit/test_profiles.py:56]: can probably remove this one.. If it doesn't return a df, other tests would fail anyways, I.E. test_result_is_not_empty.

**Review by RichardKelly84 — COMMENTED** [review]
Most of the CSV_generation can probably be removed or updated, some other small comments.

File: harvester/tests/unit/test_csv_generation.py (hunk 8 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: It will always succeed because you need to make sure that all the columns are there. its important to verify even the smallest detail to ensure that what you have is truly there.

File: harvester/tests/unit/test_csv_generation.py (hunk 9 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Same as above, gotta make sure that edge cases are addressed in tests.

File: harvester/tests/unit/test_csv_generation.py (hunk 10 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: So its a good idea to keep each of them separate because if something fails, you can know which one is the culprit right away. By smashing all of them together, you don't really have an idea as to which one failed and it defeats the purpose of having an exact test in the first place. Redundancy is better in the case of tests.

File: harvester/tests/unit/test_csv_generation.py (hunk 12 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Yes, thats the point :)

File: harvester/tests/unit/test_csv_generation.py (hunk 13 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Correct, we make sure that its within the bounds and if not then there is a problem.

File: harvester/tests/unit/test_csv_generation.py (hunk 14 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Yup, gotta make sure its within bounds.

File: harvester/tests/unit/test_csv_generation.py (hunk 15 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Correct, gotta make sure min is smaller than max.

File: harvester/tests/unit/test_csv_generation.py (hunk 16 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Gotta make sure min is smaller than max.

File: harvester/tests/unit/test_csv_generation.py (hunk 17 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Yup, gotta make sure there are records existing. In the future we would be updating the sample profiles to include new features, items and the likes, so this would catch if something is missing. Having defined test data and making sure its still valid is still a good test.

File: harvester/tests/unit/test_csv_generation.py (hunk 11 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: You're likely correct on this one, I might need to update a sample dataset with duplicates to make sure this one goes through correctly....

File: harvester/tests/unit/test_csv_generation.py (hunk 8 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_csv_generation.py (hunk 9 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_csv_generation.py (hunk 10 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_csv_generation.py (hunk 11 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_csv_generation.py (hunk 12 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_csv_generation.py (hunk 13 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_csv_generation.py (hunk 14 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_csv_generation.py (hunk 15 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_csv_generation.py (hunk 16 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_csv_generation.py (hunk 17 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_csv_generation.py]: Updated so I'm marking as resolved.

File: harvester/tests/unit/test_profiles.py (hunk 19 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_profiles.py:56]: Checking for type is likely okay here. Its a bit redundant but it won't take much time and does ensure we are working with the correct datatype.

File: harvester/tests/unit/test_ckan.py (hunk 18 above)
Comment (vmehraOCN) [reply harvester/tests/unit/test_ckan.py:75]: Will fix this now
